examples/ct/structures.go

```diff
@@ -99,15 +112,208 @@ func SignV1SCTForPrecertificate(km crypto.KeyManager, cert *x509.Certificate, t
 	keyHash := sha256.Sum256(cert.RawSubjectPublicKeyInfo)
 	precert := ct.PreCert{IssuerKeyHash: keyHash, TBSCertificate: cert.RawTBSCertificate}
 
-	timestampedEntry := ct.TimestampedEntry{Timestamp:sctInput.Timestamp, EntryType:ct.PrecertLogEntryType, PrecertEntry:precert}
-	leaf := ct.MerkleTreeLeaf{Version: ct.V1, LeafType:ct.TimestampedEntryLeafType, TimestampedEntry:timestampedEntry}
+	timestampedEntry := ct.TimestampedEntry{Timestamp: sctInput.Timestamp, EntryType: ct.PrecertLogEntryType, PrecertEntry: precert}
+	leaf := ct.MerkleTreeLeaf{Version: ct.V1, LeafType: ct.TimestampedEntryLeafType, TimestampedEntry: timestampedEntry}
 
 	return serializeAndSignSCT(km, leaf, sctInput, t)
 }
 
-func getSCTForSignatureInput(t time.Time) (ct.SignedCertificateTimestamp) {
+func getSCTForSignatureInput(t time.Time) ct.SignedCertificateTimestamp {
 	return ct.SignedCertificateTimestamp{
-		SCTVersion:ct.V1,
+		SCTVersion: ct.V1,
 		Timestamp:  uint64(t.UnixNano() / millisPerNano), // spec uses millisecond timestamps
 		Extensions: ct.CTExtensions{}}
-}
\ No newline at end of file
+}
+
+func NewLogEntry(leaf ct.MerkleTreeLeaf, certChain []*x509.Certificate) *CTLogEntry {
```

Comment:
Can you leave a TODO in here about verifying that the chain is well-formed?
By which I mean that each cert chains to the next.

Is there a plan to implement a root set restriction at some point?  That would need to be validated here too.